docs/hpc/13_tutorial_intro_shell_hpc/06_scripts_variables_loops.mdx

```diff
@@ -0,0 +1,393 @@
+-   execute (`x`): This refers to the permission to execute this file.
+
+The following table show what each of the subcolumns refer to and their possible values:
+
+| directory  | user read  | user write | user execute | group read | group write | group execute | other read | other write | other execute |
+| ---------- | ---------- | ---------- | ------------ | ---------- | ----------- | ------------- | ---------- | ----------- | ------------- |
+| `d` or `-` | `r` or `-` | `w` or `-` | `x` or `-`   | `r` or `-` | `w` or `-`  | `x` or `-`    | `r` or `-` | `w` or `-`  | `x` or `-`    |
+
+If there is a `-` in the directory column, the row refers to a file.  If it contains a `d`, the row refers to a directory.  The following columns behave in a similar manner. If they contain a `-`, the associated action is not allowed for the associated user type.
+
+### Column 2: References
+This counts the number of references (hard links) to the item (file, folder, symbolic link or “shortcut”).
+
+### Column 3: Owner
+This is the username of the user who owns the file.
+:::note
+Their permissions are indicated in the first permissions block of three after the directory column
+:::
+
+### Column 4: Group
+Each user has a primary group and is optionally a member of other groups. When a user creates a file, it is normally associated with the user’s primary group. At NYU HPC, all users are in a group named ‘users’, so group permission has little meaning.
+:::note
+Other members of this group have the permissions in the second block of three after the directory column
+:::
+
+### Column 5: Size of item
+This is the number of bytes in a file, or the number of filesystem blocks occupied by the contents of a folder.
+:::note
+We can use the `-h` option here to get a human-readable file size in megabytes, gigabytes, etc.
+:::
+
+### Column 6: Time last modified
+This is the last time the file was modified.
+
+### Column 7: Filename
+This is the name of the file/directory.
+
+## Changing Permissions
+As previously mentioned, in Unix a file has three basic permissions, each of which can be set for three types of user. Those three permission also have a numeric value:
+
+-   Read permission (“r”) - numeric value 4.
+-   Write permission (“w”) - numeric value 2.
+-   Execute permission (“x”) - numeric value 1.
+:::note
+When applied to a directory, execute permission refers to whether the directory can be entered with `cd`.
+:::
+
+You'll need to use the `chmod` command to modify permissions.  You grant permissions with `chmod who+what file` and revoke them with `chmod who-what file`. (Notice that the first has `+` and the second `-`). Here, “who” is some combination of “u”, “g”, and “o”, and “what” is some combination of “r”, “w”, and “x”.  Leaving out the `who` part of the command applies it to all user types.
+
+So, to set execute permission we use:
+```bash
+$ chmod +x demo.sh
+$ ls -l
+-rw-rw-r-- 1 yourUsername tc001 12534006 Jan 16 18:50 bash-lesson.tar.gz
```

Comment:
since our readers are our HPC users and their usernames are always their NetIDs, should we just let this be NetID?